File: Ocaml/zad2.ml
let strings = ["Ala";"ma";"kota"];;
let rec sentenceMaker (stringList, ending, separator) =
if List.length stringList = 1 then List.hd stringList^ending
else (List.hd stringList)^separator^(sentenceMaker (List.tl stringList, ending, separator));;
sentenceMaker (strings, ".", " ");;

File: Lista1/src/Lista1/zad2.scala
val strings = List()

def sentenceMaker (stringList:List[String], ending:Char, separator:Char):String =
    if (stringList == Nil) "" else
    if (stringList.length == 1) stringList.head+ending else
    stringList.head + separator + sentenceMaker(stringList.tail, ending, separator)

sentenceMaker (strings, '!', ' ')
